code/models/ShareToken.php

```diff
@@ -14,4 +21,17 @@ class ShareToken extends DataObject {
 	private static $has_one = array(
 		'Page' => 'Page'
 	);
+
+	/**
+	 * Determines whether the token is still valid (from days since it was created).
+	 *
+	 * @return bool
+	 */
+	public function isExpired() {
+		$createdSeconds = strtotime($this->Created);
+
+		$validForSeconds = (int) $this->ValidForDays * 24 * 60 * 60;
+
+		return ($createdSeconds + $validForSeconds) <= time();
```

Comment:
can you use SS_DateTime::now()? This makes it testable because then we can mock now. :D